client/src/pages/Events/Events.tsx

```diff
@@ -37,6 +46,11 @@ export function Events() {
     console.error(error);
     return <div>Error...</div>;
   }
+  const filteredEvents = futureEventsQuery.data.filter((event) =>
```

Comment:
Minor, but I suggest memoizing this since you've gone from query data, which is stable, to computed data, which is not stable across renders. I would also debounce the setter to a reasonable delay to avoid too many recomputations on filtered data